### PR TITLE
Add PlayerFlowerPotManipulateEvent

### PR DESCRIPTION
**Pull request by MisterVector**
Added an event for when a player places an item in or takes an item out of a flower pot.

This is a revisit to an old PR I had submitted nearly 3 years ago but wasn't able to finish because of an issue with server-side code that handles event cancellation.

### CODE REVIEW

**Comment by fgRuslan** [comment]
:sunflower:

**Comment by fgRuslan** [comment]
:white_flower: :tulip:

**Comment by Proximyst** [comment]
Please don't spam comments.

**Comment by fgRuslan** [comment]
Sorry

File: Spigot-API-Patches/0185-Add-PlayerFlowerPotManipulateEvent.patch
Comment (Proximyst) [inline Spigot-API-Patches/0185-Add-PlayerFlowerPotManipulateEvent.patch]:
```suggestion
+package io.papermc.paper.event.player;
```

File: Spigot-Server-Patches/0411-Implement-PlayerFlowerPotManipulateEvent.patch
Comment (Proximyst) [inline Spigot-Server-Patches/0411-Implement-PlayerFlowerPotManipulateEvent.patch]: My condolences 🤓

**Review by Proximyst — DISMISSED** [review]
Needs a rebase, but lgtm.

**Comment by MisterVector** [comment]
Alright rebased on top of latest. Had to change one line of code server-side due to latest server changes. Should be good to go!

**Review by zachbr — DISMISSED** [review]

**Comment by Proximyst** [comment]
@limbo-app ready

**Review by limbo-app — APPROVED** [review]
Pull request automatically approved for merging - a core team member has previously approved this pull request.

**Pull request merged**
